### PR TITLE
supportning non avro messages

### PR DESCRIPTION
**Pull request by stheppi**
adding support for schemaless sink record or a  record with string schema and json payload

### CODE REVIEW

File: kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala [hunk 1]
```diff
@@ -19,28 +19,136 @@ package com.datamountaineer.streamreactor.connect.influx.writers
...
             settings: InfluxSettings): BatchPoints = {
+
+    def handleSinkRecord(record: SinkRecord): Option[Point] = {
+
+      Option(record.valueSchema()) match {
+        case None =>
+          //try to take it as string
+          record.value() match {
+            case map: java.util.Map[_, _] => buildPointFromMap(record, settings)
+            case _ => sys.error("For schemaless record only String and Map types are supported")
```
Comment (Antwnis) [inline kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala]: How about Int | Long | Boolean etc ?

Comment (Antwnis) [inline kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala]: better say

`and Map[String, Any] types are supported`

File: kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala [hunk 2]
```diff
@@ -19,28 +19,136 @@ package com.datamountaineer.streamreactor.connect.influx.writers
...
+        case (field, value: Long) => builder.addField(field, value)
+        case (field, value: Int) => builder.addField(field, value)
+        case (field, value: BigInt) => builder.addField(field, value)
+        case (field, value: Byte) => builder.addField(field, value)
+        case (field, value: Short) => builder.addField(field, value)
+        case (field, value: Double) => builder.addField(field, value)
+        case (field, value: Float) => builder.addField(field, value)
+        case (field, value: Boolean) => builder.addField(field, value)
+        case (field, value: java.math.BigDecimal) => builder.addField(field, value)
+        case (field, value: String) => builder.addField(field, value)
```
Comment (Antwnis) [inline kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala]: shouldn't you have

```case (field, value: Map[String, Any) => ..```

To allow nested Json messages ?

File: kafka-connect-influxdb/src/test/scala/com/datamountaineer/streamreactor/connect/influx/InfluxBatchPointsBuilderFnTest.scala [hunk 3]
```diff
@@ -0,0 +1,757 @@
...
+    val map = PointMapFieldGetter.fields(point)
+    map.size shouldBe 3
+
+    map.get("_id") shouldBe "580151bca6f3a2f0577baaac"
+    map.get("this_is_renamed") shouldBe "Clements Crane"
+    map.get("email") shouldBe "clements.crane@terragen.io"
+
+  }
+
+  "throw an error of if nested json since there is no flattening of json for a sink record with string json payload" in {
```
Comment (Antwnis) [inline kafka-connect-influxdb/src/test/scala/com/datamountaineer/streamreactor/connect/influx/InfluxBatchPointsBuilderFnTest.scala]: I would object to that. We should provide a mechanism like

```flattenJson = '.'```

In that case we would automatically flatten the json structure using this separator

i.e. 

```  """
         | {
         |    "eyeColor": "brown",
         |    "name.first": "Christian",
         |    "name.last": "Melton"
         |    }
         | }```

File: kafka-connect-influxdb/src/test/scala/com/datamountaineer/streamreactor/connect/influx/InfluxBatchPointsBuilderFnTest.scala [hunk 4]
```diff
@@ -0,0 +1,757 @@
...
+      InfluxBatchPointsBuilderFn(Seq(record), settings)
+    }
+  }
+
+  "throw an error of if array is present in json since there is no flattening of json for a sink record with string json payload" in {
+    val jsonPayload =
+      """
+        | {
+        |    "eyeColor": "brown",
+        |     "tags": [
```
Comment (Antwnis) [inline kafka-connect-influxdb/src/test/scala/com/datamountaineer/streamreactor/connect/influx/InfluxBatchPointsBuilderFnTest.scala]: Here we could be flattening using the javascript style.. i.e.

```tags[0] = "ut",
     tags[1] = "dolor" ...
```

File: kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala (hunk 1 above)
Comment (stheppi) [inline kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala]: Ant, take a look at the else branch. if you don't have a schema, and you send a json and use JsonConverter you get back a java.util.Map.  the logic there is correct. If you send an int/long/Boolean what would you insert it against in influx?

File: kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala (hunk 2 above)
Comment (stheppi) [inline kafka-connect-influxdb/src/main/scala/com/datamountaineer/streamreactor/connect/influx/writers/InfluxBatchPointsBuilderFn.scala]: What if the nested has fields already processed? basically you do an ovewrite? For now we don't support them

File: kafka-connect-influxdb/src/test/scala/com/datamountaineer/streamreactor/connect/influx/InfluxBatchPointsBuilderFnTest.scala (hunk 3 above)
Comment (stheppi) [inline kafka-connect-influxdb/src/test/scala/com/datamountaineer/streamreactor/connect/influx/InfluxBatchPointsBuilderFnTest.scala]: I could do that

**Pull request merged**